### PR TITLE
Update RLS, Rustfmt, and Clippy

### PR DESCRIPTION
**Pull request by nrc**
Fixes bustage due to https://github.com/rust-lang/rust/pull/52332

r? @kennytm

### CODE REVIEW

File: src/Cargo.lock [hunk 1]
```diff
@@ -1636,8 +1636,8 @@ dependencies = [
 
 [[package]]
 name = "racer"
-version = "2.1.2"
-source = "registry+https://github.com/rust-lang/crates.io-index"
+version = "2.1.3"
+source = "git+https://github.com/racer-rust/racer?rev=2af3ab88420e968500fcfc75418be7e852debfa0#2af3ab88420e968500fcfc75418be7e852debfa0"
```
Comment (kennytm) [inline src/Cargo.lock]: Could we depend on the [released](https://crates.io/crates/racer) racer 2.1.3? Using a git dependency will fail tidy.

Comment (nrc) [reply src/Cargo.lock]: :-(

**Comment by kennytm** [comment]
You need to update Clippy as well (RLS depends on `clippy_lints`).

```
[01:10:20] error[E0261]: use of undeclared lifetime name `'a`
[01:10:20]    --> tools/clippy/clippy_lints/src/write.rs:220:32
[01:10:20]     |
[01:10:20] 220 | fn check_tts(cx: &EarlyContext<'a>, tts: &ThinTokenStream, is_write: bool) -> Option<String> {
[01:10:20]     |                                ^^ undeclared lifetime
[01:10:20]
[01:10:20] error: aborting due to previous error
```

**Comment by kennytm** [comment]
@oli-obk  Clippy is still test fail for the following two tests (also, needs to rebase):

```
[01:02:22] failures:
[01:02:22]     [ui] ui/infinite_iter.rs
[01:02:22]     [ui] ui/range.rs
[01:02:22] 
```

**Comment by kennytm** [comment]
@bors r- retry 

```
[01:13:00] error[E0599]: no variant named `GlobalAsm` found for type `rustc::hir::def::Def` in the current scope
[01:13:00]    --> tools/clippy/clippy_lints/src/utils/mod.rs:942:9
[01:13:00]     |
[01:13:00] 942 |         Def::GlobalAsm(id) => Some(id),
[01:13:00]     |         ^^^^^^^^^^^^^^^^^^ variant not found in `rustc::hir::def::Def`
```

**Comment by oli-obk** [comment]
Uhm. I fixed that

**Comment by oli-obk** [comment]
The clippy submodule is fine. ia rls pointing at the wrong thing and the override isn't working anymore?

**Comment by nrc** [comment]
Now with updated Clippy too (I think I updated before @oli-obk 's clippy fix landed)

**Comment by alexcrichton** [comment]
@nrc if you get a chance, mind throwing [this diff in as well](https://gist.github.com/alexcrichton/8790714ad5972f8a10dad4aa247bf662) to prevent https://github.com/rust-lang/rust/issues/53006 from happening again as well? (commenting out the `rustc-ap-syntax` line too)

**Comment by kennytm** [comment]
Still clippy failure 😥 

```
[00:55:15] error[E0053]: method `visit_path` has an incompatible type for trait
[00:55:15]    --> tools\clippy\clippy_lints\src\utils\internal_lints.rs:201:5
[00:55:15]     |
[00:55:15] 201 |     fn visit_path(&mut self, path: &'tcx Path, _: NodeId) {
[00:55:15]     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `rustc::hir::HirId`, found struct `syntax::ast::NodeId`
[00:55:15]     |
[00:55:15]     = note: expected type `fn(&mut utils::internal_lints::LintCollector<'a, 'tcx>, &'tcx rustc::hir::Path, rustc::hir::HirId)`
[00:55:15]                found type `fn(&mut utils::internal_lints::LintCollector<'a, 'tcx>, &'tcx rustc::hir::Path, syntax::ast::NodeId)`
[00:55:15] 
[00:55:15] [RUSTC-TIMING] failure test:false 1.718
[00:55:15]    Compiling rls-analysis v0.14.0
[00:55:15] error[E0053]: method `visit_path` has an incompatible type for trait
[00:55:15]    --> tools\clippy\clippy_lints\src\use_self.rs:211:5
[00:55:15]     |
[00:55:15] 211 |     fn visit_path(&mut self, path: &'tcx Path, _id: NodeId) {
[00:55:15]     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `rustc::hir::HirId`, found struct `syntax::ast::NodeId`
[00:55:15]     |
[00:55:15]     = note: expected type `fn(&mut use_self::UseSelfVisitor<'a, 'tcx>, &'tcx rustc::hir::Path, rustc::hir::HirId)`
[00:55:15]                found type `fn(&mut use_self::UseSelfVisitor<'a, 'tcx>, &'tcx rustc::hir::Path, syntax::ast::NodeId)`
[00:55:15] 
[00:55:15] error: aborting due to 2 previous errors
```

**Comment by nrc** [comment]
And another Clippy update and Alex's patch

@bors: r+

**Pull request merged**